macros/SXCmacros.cfg

```diff
@@ -6307,13 +6296,38 @@ $maptext|"
       amount=-$gold_die
       side=$dying_friend.side
     [/gold]
-     [/else]
-    [/if
+    [/else]
+    [/if]
     {CLEAR_VARIABLE gold_die}
     {CLEAR_VARIABLE dying_friend}
     {CLEAR_VARIABLE random}
     {CLEAR_VARIABLE i}
   [/event]
+  
+   [event]
+    name=new turn
+    first_time_only=no
+    {FOREACH dead i}
+      [store_gold]
+        side=$dead[$i|].side
+        variable=side_gold
+      [/store_gold]
+      [gold]
+        side=$dead[$i|].side
+        amount=-$side_gold
+      [/gold]
+      [gold]
+        side=$dead[$i|].side
+        amount=$dead[$i|].variables.gold
+      [/gold]
+      [unstore_unit]
+        variable=dead[$i|]
+        find_vacant=yes
+      [/unstore_unit]
+      {CLEAR_VARIABLE side_gold}
+    {NEXT i}
+    {CLEAR_VARIABLE dead}
+  [/event]
```

Comment:
Please rebase the branch, because parts of this review have already been merged, for example the whole revive mode.